apps/api/test/functional/certificate.spec.ts

```diff
@@ -0,0 +1,115 @@
+import { stub } from "@test/services/stub";
+import { WalletTestingService } from "@test/services/wallet-testing.service";
+
+jest.setTimeout(30000);
+
+describe("Certificate API", () => {
+  const userRepository = container.resolve(UserRepository);
+  const apiKeyRepository = container.resolve(ApiKeyRepository);
+  const dbService = container.resolve(DbTestingService);
+  const walletService = new WalletTestingService(app);
+  const userWalletRepository = container.resolve(UserWalletRepository);
+  let apiKeyGenerator: ApiKeyGeneratorService;
+  let config: jest.Mocked<CoreConfigService>;
+
+  async function createTestUser() {
+    const { user, token, wallet } = await walletService.createUserAndWallet();
+    const userWithId = { ...user, userId: faker.string.uuid() };
+    config = stub<CoreConfigService>({ get: jest.fn() });
+    config.get.mockReturnValue("test");
+    apiKeyGenerator = new ApiKeyGeneratorService(config);
+    const apiKey = apiKeyGenerator.generateApiKey();
+    const obfuscatedKey = apiKeyGenerator.obfuscateApiKey(apiKey);
+
+    jest.spyOn(userRepository, "findById").mockImplementation(async id => {
+      if (id === userWithId.userId) {
+        return {
+          ...userWithId,
+          trial: false,
+          userWallets: { isTrialing: false }
+        };
+      }
+      return undefined;
+    });
+
+    jest.spyOn(apiKeyRepository, "findOneBy").mockImplementation(async key => {
+      if (key.keyFormat === obfuscatedKey) {
+        const now = new Date().toISOString();
+        return {
+          id: faker.string.uuid(),
+          userId: userWithId.userId,
+          key: apiKey,
+          hashedKey: await apiKeyGenerator.hashApiKey(apiKey),
+          keyFormat: "sk",
+          name: "test",
+          createdAt: now,
+          updatedAt: now,
+          expiresAt: null,
+          lastUsedAt: null
+        };
+      }
+      return undefined;
+    });
+
+    jest.spyOn(userWalletRepository, "findOneByUserId").mockImplementation(async id => {
+      if (id === user.id) {
+        return {
+          ...wallet,
+          address: wallet.address
+        };
+      }
+    });
+
+    return { user: userWithId, token, apiKey, wallet };
+  }
+
+  beforeEach(async () => {
+    await dbService.cleanAll();
+  });
+
+  describe("POST /v1/certificate/create", () => {
+    it("creates a certificate for the authenticated user", async () => {
+      const { apiKey, wallet } = await createTestUser();
+
+      const response = await app.request("/v1/certificates", {
+        method: "POST",
+        headers: new Headers({
+          "Content-Type": "application/json",
+          "x-api-key": apiKey
+        })
+      });
+
+      expect(response.status).toBe(200);
+      const result = (await response.json()) as CreateCertificateResponse;
+
+      expect(result.data?.certPem).toBeDefined();
+      expect(result.data?.pubkeyPem).toBeDefined();
+      expect(result.data?.encryptedKey).toBeDefined();
```

Comment:
non-blocking: this could prob be tested a bit further. like that's a defined string:
```suggestion
      expect(result.data).toMatchObject({
        certPem: expect.any(String),
        pubkeyPem: expect.any(String),
        encryptedKey: expect.any(String)
      });
```

Reply:
Will do in another branch